app/src/main/java/space/taran/arkshelf/data/LinkLocalDataSource.kt

```diff
@@ -23,65 +24,59 @@ import kotlin.io.path.createTempFile
 import java.security.MessageDigest
 import java.util.zip.ZipFile
 import javax.inject.Inject
+import javax.inject.Named
 import kotlin.io.path.createDirectories
 import kotlin.io.path.extension
 import kotlin.io.path.getLastModifiedTime
 import kotlin.io.path.isDirectory
-
-private class LinkWithPath(val link: Link, val path: Path)
+import kotlin.io.path.notExists
 
 class LinkLocalDataSource @Inject constructor(
     private val preferences: UserPreferences,
-    private val appCtx: Context
+    private val appCtx: Context,
+    @Named("PAGE_SIZE") private val pageSize: Int,
 ) {
     private val klaxon = Klaxon()
-    private val latestByFolder = mutableMapOf<Path, MutableList<LinkWithPath>>()
-    private val loadMoreMutex = Mutex()
-    private val cacheDir = appCtx.cacheDir.resolve(CACHE_IMAGES_DIR).toPath().createDirectories()
+    private val cacheDir =
+        appCtx.cacheDir.resolve(CACHE_IMAGES_DIR).toPath().createDirectories()
 
     fun createLinkFile(link: Link, basePath: Path) {
         val jsonFile = generateJsonFile(link)
         val savePath = basePath.resolve(Path("${sha512(link.url)}.link"))
         zipFiles(jsonFile, link.imagePath, savePath)
     }
 
-    suspend fun loadMore(): List<Link> = withContextAndLock(
-        Dispatchers.Default,
-        loadMoreMutex
-    ) {
-        val linkFolder = preferences.getLinkFolder()
-            ?: return@withContextAndLock emptyList()
+    suspend fun loadFiles(page: Int): Pair<List<Link>, Boolean> = withContext(Dispatchers.Default) {
```

Comment:
Let's replace `Pair<List<Link>, Boolean>` with dedicated structure for the sake of readability.